api/app/signals/apps/services/domain/dsl.py

```diff
@@ -82,10 +82,17 @@ class SignalDslService(DslService):
     context_func = SignalContext()
     signal_manager = SignalManager()
 
-    def process_routing_rules(self, signal):
+    def process_routing_rules(self, signal):  # noqa C901
         ctx = self.context_func(signal)
         rules = RoutingExpression.objects.select_related('_expression', '_department')
         for rule in rules.filter(is_active=True, _expression___type__name='routing').order_by('order'):
+            if (rule._department and rule._user
+                    and not rule._user.profile.departments.filter(id=rule._department.id).exists()):
+                # The user is no longer a member of the department so let's de activate the rule
+                rule.is_active = False
```

Comment:
I like this way of dealing with users whose association with some department ended. Do we need something similar for users that are no longer active but still associated with aforementioned department?

Reply:
That is also a good one to add. Will do that!